### PR TITLE
Update tarn.js

### PR DESCRIPTION
**Pull request by kibertoad**

### CODE REVIEW

**Comment by kibertoad** [comment]
@elhigu @koskimas  `.afterCreate()` hook explodes as unsupported now. We don't support and don't want to support that anymore? Listening to creation events is the way to go now?

**Comment by elhigu** [comment]
afterCreate is implemented in knex side and looks like it is passed to pool paameters by accident.

File: test/unit/seed/seeder.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ describe('Seeder._waterfallBatch', function() {
   it('should throw an error with correct file name', (done) => {
     seeder._waterfallBatch(['1-first.js', '2-second.js']).catch((error) => {
       expect(error.message).to.match(
-        /^Error while executing "(\/\w+)+\/1-first\.js" seed: throwing in first file$/
+        /^Error while executing .*1-first.js" seed: throwing in first file$/
```
Comment (kibertoad) [inline test/unit/seed/seeder.js]: Original assertion was failing on Windows

File: lib/client.js [hunk 2]
```diff
@@ -297,7 +297,15 @@ Object.assign(Client.prototype, {
       return;
     }
 
-    this.pool = new Pool(this.getPoolSettings(config.pool));
+    const tarnPoolConfig = {
+      ...this.getPoolSettings(config.pool),
+    };
+    // afterCreate is an internal knex param, tarn.js does not support it
+    if (tarnPoolConfig.afterCreate) {
+      delete tarnPoolConfig.afterCreate;
```
Comment (alexandrebodin) [inline lib/client.js]: You will have the same issue with `beforeDestroy` I think.

Comment (elhigu) [reply lib/client.js]: `beforeDestroy` is deprecated and is not really useful hook anyways (it has been deprecated already for a long time). I think we can drop it already.

Comment (alexandrebodin) [reply lib/client.js]: Sure the PR should simply delete it from the code so there is no way it is passed by error then :)

Comment (kibertoad) [reply lib/client.js]: Removed `beforeDestroy` from typings and implementation, and mentioned in migration guide that it's no longer supported. Users should still update their code not to pass it, though, as silently ignoring configuration options is confusing.

Comment (elhigu) [reply lib/client.js]: When people start complaining that their hacky code (*) doesn't work without it we can tell them to use pool's event handlers instead.

*)  code relaying to that hook cannot work in a  robust manner unless it is counting more or less how many connections were discarded from pool... so far all the people who has been using it and with whom I've been talking to have made invalid assumptions when that is called and have tried to do some DB disconnected kind recognitions and knex re-inits there.

Comment (kibertoad) [reply lib/client.js]: Yeah, migration guide already mentions pool event handlers.

**Pull request merged**